File: app/src/main/java/me/manabu/Api/ApiInterface.java
package me.manabu.Api;

import java.util.List;

import me.manabu.Api.Models.DeckModel;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ApiInterface {

    @POST("/api/v1/auth/login")
    Call<ResponseBody> login(@Body String gToken);

    @GET("/api/v1/auth/login")
    Call<ResponseBody> getAllDecks(@Body String body);

    @GET("/deck/{id}/lessons")
    Call<List<DeckModel>> getLessonsFromDeck(@Path("id") int id);
}

File: app/src/main/java/me/manabu/webapi/ApiInterface.kt
package me.manabu.webapi

import me.manabu.webapi.models.BasicResponse
import me.manabu.webapi.models.DeckLevelModel
import me.manabu.webapi.models.DeckModel
import me.manabu.webapi.models.UserDeckModel
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiInterface {

    @GET("/api/v1/decks/get")
    fun getAllDecks(): Call<BasicResponse<List<DeckModel>>>

    @GET("/api/v1/deck_to_user/{deckId}/{userId}")
    fun copyDeckToUser(@Path("deckId") deckId: String, @Path("userId") userId: String): Call<ResponseBody>

    @GET("/api/v1/user_deck/delete/{deckId}")
    fun removeUserDeck(@Path("deckId") deckId: String): Call<ResponseBody>

    @GET("/api/v1/user_decks/get/{userId}")
    fun getUserDecks(@Path("userId") userId: String): Call<BasicResponse<List<UserDeckModel>>>

    @GET("/api/v1/deck/{deckId}/lessons")
    fun getAvailableLessons(@Path("deckId") deckId: String): Call<BasicResponse<List<DeckLevelModel>>>

}